python/hsfs/hopsworks_udf.py

```diff
@@ -31,7 +32,18 @@
 from hsfs.transformation_statistics import TransformationStatistics
 
 
-def udf(return_type: Union[List[type], type]) -> "HopsworksUdf":
+class UDFType(Enum):
```

Comment:
This is actually transformation type. The UDF is actually the same and can be used for both ODT and MDT. I understand this type is needed for the udf to return different column names, but imo this is better to be done in `transformation_function` class and also the validation of the transformation type. But it is not critical, can be refactored later.

Reply:
You are right. I will move it out to transformation function class.